Library/Homebrew/cask/cask_loader.rb

```diff
@@ -299,7 +299,12 @@ def load(config:)
 
         json_cask = Homebrew::API.merge_variations(json_cask).deep_symbolize_keys.freeze
 
-        cask_options[:tap] = Tap.fetch(json_cask[:tap]) if json_cask[:tap].to_s.include?("/")
+        # We could probably just default to the core cask tap in all cases without problems.
+        cask_options[:tap] = if Homebrew::API.internal_json_v3?
+          CoreCaskTap.instance
+        elsif json_cask[:tap].to_s.include?("/")
+          Tap.fetch(json_cask[:tap])
+        end
```

Comment:
Does anyone know if we still need the logic? It seems like it's probably necessary since the only tap I'd expect to see here would be the core cask tap.